ToDo.Core/ViewModels/ToDoViewModel.cs

```diff
@@ -111,35 +199,72 @@ internal async Task AddNewTodo()
             await _navigationService.Navigate<AddViewModel>();
         }
 
-        internal async Task RemoveTodo(Models.ToDo toDoItem)
+        internal async Task RemoveTodo(VisualElement frontSide)
         {
-            var confirmed = await _userDialogService.ConfirmAsync(new ConfirmConfig()
-            {
-                Message = "Are you sure you want to delete this item?",
-                OkText = "Delete",
-                CancelText = "Cancel"
-            });
+            //var confirmed = await _userDialogService.ConfirmAsync(new ConfirmConfig()
+            //{
+            //    Message = "Are you sure you want to delete this item?",
+            //    OkText = "Delete",
+            //    CancelText = "Cancel"
+            //});
 
-            if (confirmed)
-            {
-                ToDos.Remove(toDoItem);
+            //if (confirmed)
+            //{
+            //    ToDos.Remove(SelectedTodo);
+
+            //    await _todoService.DeleteTodo(SelectedTodo);
+            //}
+
+            var frontSideToFlip = frontSide.FindByName<Grid>("FrontSideToFlip");
+
+            var parent = (Grid)frontSide.Parent;
+            var backSide = parent.FindByName<Grid>("BackSide");
+            var backSideToFlip = frontSide.FindByName<Grid>("BackSideToFlip");
+
+            await Flip(frontSideToFlip, backSideToFlip);
 
-                await _todoService.DeleteTodo(toDoItem);
-            }
         }
 
-        internal async Task EditTodo(Models.ToDo toDoItem)
+        internal async Task Flip(VisualElement from, VisualElement to)
         {
-            await _navigationService.Navigate<EditViewModel, Models.ToDo>(toDoItem);
+            await from.RotateYTo(-90, 300, Easing.SpringIn);
+            to.RotationY = 90;
+            FadeBackgroundBackSideVisibility = true;
+            FadeBackgroundFrontSideVisibility = false;
+            from.RotationY = 0;
+            await to.RotateYTo(0, 300, Easing.SpringOut);
         }
 
-        internal async Task ChangeState(Models.ToDo toDoItem)
+        internal async Task DeleteTodo()
         {
-            toDoItem.State = (toDoItem.State == State.Done) ? State.Pending : State.Done;
+            ToDos.Remove(SelectedTodo);
 
-            await _todoService.SaveTodoAsync(toDoItem);
+            await _todoService.DeleteTodo(SelectedTodo);
 
-            await LoadTodos();
+            FadeBackgroundBackSideVisibility = false;
```

Comment:
The boolean property can stay in the viewmodel, even though it is animation-specific.
THis is because it does not reference the view, but only a simple type (bool)